### PR TITLE
Fix: max-len will warn indented comment lines (fixes #6322)

### LINKED ISSUE #6322: max-len indented comments
**What version of ESLint are you using?** `2.11.1` (from my package.json: ^2.11.1)

**What parser (default, Babel-ESLint, etc.) are you using?** `default`

**Please show your full configuration:** [full config](https://github.com/tbremer/eslint-config-tbremer/blob/master/index.js) piece concerned with:
```json
"max-len": [
  2,
  {
    "code": Infinity,
    "comments": 80,
    "tabWidth": 2
  }
],
```

**What did you do? Please include the actual source code causing the issue.** I found the issues when a single line comment was indented but longer than my set `comments` length

```javascript
function foo(something) {
  if (!something) throw new Error('something is reqquired');
  
  // this is a really long commnet this is a really long commnet this is a really long commnet this is a really long commnet
  return somethingDifferent;
}
```

_actual_ code with issue:

```javascript
export default class Maitre {
  set port(p) {
    if (this.__port__ !== undefined) throw new Error('Port should not be reassigned.');

    //this is a long comment this is a long comment this is a long comment this is a long comment this is a long commentthis is a long comment this is a long comment this is a long comment this is a long comment this is a long commentthis is a long comment this is a long comment this is a long comment this is a long comment this is a long comment
    this.__port__ = p;
  }
}
```

**What did you expect to happen?** max-len to throw an error

**What actually happened? Please include the actual, raw output from ESLint.** went a long like everything was cool.

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

File: lib/rules/max-len.js [hunk 1]
```diff
@@ -157,10 +157,12 @@ module.exports = {
          */
         function isFullLineComment(line, lineNumber, comment) {
             var start = comment.loc.start,
-                end = comment.loc.end;
+                end = comment.loc.end,
+                isFirstTokenOnLine = !line.slice(0, comment.loc.start.column).trim();
```
Comment (kaicataldo) [inline lib/rules/max-len.js]: Reusing this from the no-inline-comments rule https://github.com/eslint/eslint/blob/cd1b0574f1af79d189259f4e1a255230d24b37ee/lib/rules/no-inline-comments.js#L39

File: lib/rules/max-len.js [hunk 2]
```diff
@@ -157,11 +157,14 @@ module.exports = {
          */
         function isFullLineComment(line, lineNumber, comment) {
             var start = comment.loc.start,
-                end = comment.loc.end;
+                end = comment.loc.end,
+                isFirstTokenOnLine = !line.slice(0, comment.loc.start.column).trim(),
```
Comment (kaicataldo) [inline lib/rules/max-len.js]: Reusing this from the no-inline-comments rule https://github.com/eslint/eslint/blob/cd1b0574f1af79d189259f4e1a255230d24b37ee/lib/rules/no-inline-comments.js#L39

Comment (platinumazure) [inline lib/rules/max-len.js]: That's a pretty arcane bit of code- would it be worth extracting that to a place like ast-utils?

Comment (kaicataldo) [inline lib/rules/max-len.js]: Could be - it's pretty niche (really only useful to see if the comment is the first thing on that line)

Comment (kaicataldo) [inline lib/rules/max-len.js]: Other rules (such as [lines-around-comment](https://github.com/eslint/eslint/blob/master/lib/rules/lines-around-comment.js#L148) are able to use `sourceCode.getTokenBefore` and could check the location that way - in this rule I only had access to the line, so I thought this was a pretty elegant solution

**Comment by ilyavolodin** [comment]
LGTM

**Pull request merged**